### PR TITLE
fix: dev: Calling super in init.

### PR DESCRIPTION
**Pull request by ocelotl**
Adding calls for ``super`` in several ``__init__`` methods that were missing it.

### CODE REVIEW

File: lib/topology/platforms/node.py [hunk 1]
```diff
@@ -246,6 +246,7 @@ class BaseNode(HighLevelShellAPI, LowLevelShellAPI, ServicesAPI, StateAPI):
 
     @abstractmethod
     def __init__(self, identifier, **kwargs):
+        super(BaseNode, self).__init__()
```
Comment (dajose) [inline lib/topology/platforms/node.py]: how does this work (given multiple parent classes) ?

File: lib/topology/interact.py [hunk 2]
```diff
@@ -53,6 +53,7 @@ class NamespaceCompleter(object):
     """
 
     def __init__(self, ns):
+        super(NamespaceCompleter, self).__init__()
```
Comment (dajose) [inline lib/topology/interact.py]: I remember our discussion... but what comes with calling the base `object` `__init__ `?

File: lib/topology/platforms/node.py (hunk 1 above)
Comment (ocelotl) [inline lib/topology/platforms/node.py]: @dajose run this:



```
class Susana(object):
    def __init__(self):
        print('Susan')
        # super(Susana, self).__init__()


class Laura(Susana):
    def __init__(self):
        print('Laura')
        super(Laura, self).__init__()


class Francisca(object):
    def __init__(self):
        print('Susan')
        super(Francisca, self).__init__()


class Tamara(Francisca):
    def __init__(self):
        print('Tamara')
        super(Tamara, self).__init__()


class Maria(object):
    def __init__(self):
        print('Maria')
        super(Maria, self).__init__()


class Natasha(Laura, Maria):
    def __init__(self):
        print('Natasha')
        super(Natasha, self).__init__()


class Fernanda(Tamara, Maria):
    def __init__(self):
        print('Fernanda')
        super(Fernanda, self).__init__()


print('Creating Natasha')
Natasha()
print()
print('Creating Fernanda')
Fernanda()
```

Now, uncomment ``# super(Susana, self).__init__()``, run again.

``super`` does not just call the matching parent class method, it calls the *next* matching method in the parent classes following the MRO.

Having ``super`` called in all ``__init__`` methods is necesary precisely for these situations of multiple inheritance. If one of those multiple parent classes does not call ``super`` in its ``__init__`` (``Susana``) it breaks the chain of calls, making it impossible for the class that has multiple parents (``Natasha``) to receive the attributes introduced by the second parent class (``Maria``).

File: lib/topology/interact.py (hunk 2 above)
Comment (ocelotl) [inline lib/topology/interact.py]: Preserving the chain of calls in multiple inheritance situations.

**Pull request merged**